CMRF/Core/Core.php

```diff
@@ -22,13 +27,53 @@ public function getFactory() {
 
   protected abstract function getConnection($connector_id);
 
-  public function createCall($connector_id, $entity, $action, $parameters, $options = NULL, $callback = NULL) {
-    return $this->callfactory->createOrFetch($connector_id,$this,$entity,$action,$parameters,$options,$callback);
+  /**
+   * @param string $connector_id
+   * @param string $entity
+   * @param string $action
+   * @param array $parameters
+   * @param array|null $options
+   * @param $callback
+   * @param string $api_version
+   *
+   * @return \CMRF\Core\Call
+   *
+   * @deprecated Use createCallV3() or createCallV4() instead.
+   */
+  public function createCall($connector_id, $entity, $action, $parameters, $options = NULL,
+    $callback = NULL/*, string $api_version = '3'*/
+  ) {
+    if (6 === func_num_args()) {
+      $api_version = '3';
+    } else {
+      $api_version = func_get_arg(6);
+    }
```

Comment:
This does not seem to work reliably when one of the optional arguments is not given. Why is this necessary in the first place? Let's just add the `$api_version` parameter with a default value of `3` for backwards compatibility.

I'm also not sure whether deprecating this is a good idea since this is being called a lot in implementations. Leaving it as-is would also save us the version-specific methods further below.

Reply:
Why do you think it does not work reliable? It is of course not necessary to deprecate the method and the additional argument could be added to the method signature in a later release. Doing it this way would give implementors of subclasses that override the method time to change their signature. If we can make a BC immediately it definitely would be fine :)

Reply:
`func_num_args()` returns the number of arguments *passed* to the method, so we can not assume it always returns `6` when the `$api_version` parameter is not passed, because it could also be `5` or `4` due to the last two parameters being optional.

You're right that changing the signature would be breaking implementations overriding this method, so if we change the signature we'd have to release this with `3.0.0` instead of `2.1.0` as is planned. Let's make `func_num_args()` check for `< 7` instead. I'll push a change for that.